pkgs/top-level/all-packages.nix

```diff
@@ -89,7 +89,13 @@ with pkgs;
     { deps = [ autoconf264 automake111x gettext libtool ]; }
     ../build-support/setup-hooks/autoreconf.sh;
 
-  autoPatchelfHook = makeSetupHook { name = "auto-patchelf-hook"; }
+  autoPatchelfHook = makeSetupHook {
+    name = "auto-patchelf-hook";
+    substitutions = rec {
+      ld = "${stdenv.cc}/nix-support/dynamic-linker";
+      ld32 = if stdenv.hostPlatform.is32bit then ld
+        else "${stdenv_32bit.cc}/nix-support/dynamic-linker-m32";
```

Comment:
A shortcut in `stdenv.cc.bintools` (like the existing `dynamicLinker`) would remove the reliance on this output, as it's probably only supported for GCC libc.